cmd/run/run.go

```diff
@@ -302,19 +314,11 @@ func MustDefaultConfigWithRandomPorts() *Config {
 	config.Playground.Enabled = false
 	config.Metrics.Enabled = false
 
-	l, err := net.Listen("tcp", "")
-	if err != nil {
-		panic(err)
-	}
-	defer l.Close()
-	httpPort := l.Addr().(*net.TCPAddr).Port
+	httpPort, httpPortReleaser := TCPRandomPort()
 
-	l, err = net.Listen("tcp", "")
-	if err != nil {
-		panic(err)
-	}
-	defer l.Close()
-	grpcPort := l.Addr().(*net.TCPAddr).Port
+	grpcPort, grpcPortReleaser := TCPRandomPort()
+	httpPortReleaser()
```

Comment:
Should we defer these so that we know there isn't a race on the port being acquired before the return of this function?

```go
defer func() {
    httpPortReleaser()
    grpcPortReleaser()
}()
```

Reply:
updated